src/helpers.js

```diff
@@ -3,8 +3,100 @@ export const HELPERS_ID = '\0commonjsHelpers';
 export const HELPERS = `
 export var commonjsGlobal = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : {};
 
-export function commonjsRequire () {
-	throw new Error('Dynamic requires are not currently supported by rollup-plugin-commonjs');
+const realRequire = require;
+
+// Avoid evaluating when not necessary (requiring here will mainly break tests).
+let pathModule;
+
+const DYNAMIC_REQUIRE_LOADER_MAP = new Map();
+const CACHE = Object.create(null); // This is how it's done in real 'require'
+const DEFAULT_PARENT_MODULE = {id: '<' + 'rollup' + '>', exports: {}, parent: undefined, filename: null, loaded: false, children: [], paths: []};
+
+export function commonjsRequire (path, originalModuleDir) {
+    // Transform path to distribution path
+    if (!pathModule)
+        pathModule = realRequire('path');
+
+    const isRelative = /[\\/\\\\]/.test(path);
+    
+    let relPath;
+    
+    while (true) {
+        if (isRelative) {
+            relPath = pathModule.normalize(pathModule.join(originalModuleDir || '', path)).replace(/\\\\/g, '/');
+        } else {
+            if (originalModuleDir)
+                relPath = pathModule.normalize(pathModule.join(originalModuleDir || '', '../node_modules/', path)).replace(/\\\\/g, '/');
```

Comment:
The `originalModuleDir || ''` check is probably not necessary inside the if statement ;)